src/game/bank/bank.cpp

```diff
@@ -108,8 +110,17 @@ bool Bank::transferTo(const std::shared_ptr<Bank> destination, uint64_t amount)
 			return false;
 		}
 
-		if (destinationPlayer->getTown()->getID() < g_configManager().getNumber(MIN_TOWN_ID_TO_BANK_TRANSFER, __FUNCTION__)) {
-			g_logger().warn("Bank::transferTo: denied town: {}", destinationPlayer->getTown()->getID());
+		const auto destinationTownId = destinationPlayer->getTown()->getID();
+		const auto bankableTownId = bankablePlayer->getTown()->getID();
+		const auto minTownIdToTransferFromMain = g_configManager().getNumber(MIN_TOWN_ID_TO_BANK_TRANSFER_FROM_MAIN, __FUNCTION__);
+
+		if (destinationTownId < minTownIdToTransferFromMain && bankableTownId >= minTownIdToTransferFromMain) {
+			g_logger().warn("Bank::transferTo: denied town: {}", destinationTownId);
```

Comment:
try to improve this message like: ("[{}] Player {} is not from main town, trying to transfer money to player {} in {} town", __FUNCTION__, playerName, playerDest, destinationTownId)